src/cli/main.rs

```diff
@@ -447,17 +470,26 @@ fn format(opt: opt::Opt) -> Result<i32> {
 
                         let tx = tx.clone();
                         pool.execute(move || {
-                            tx.send(
-                                format_file(&path, config, range, &opt, verify_output).map_err(
-                                    |error| {
-                                        ErrorFileWrapper {
-                                            file: path.display().to_string(),
-                                            error,
-                                        }
-                                        .into()
-                                    },
-                                ),
-                            )
+                            #[cfg(not(feature = "editorconfig"))]
+                            let c = config;
+                            #[cfg(feature = "editorconfig")]
+                            let c = is_default_config
+                                .then(|| {
+                                    editorconfig::parse(Some(config), &path)
+                                        .expect("could not parse editorconfig")
```

Comment:
Rather than `.expect`'ing here, we should propogate it back up (through the `tx.send`) so its handled nicely using `anyhow`, and also add in some context. This is mainly because it also appropriately JSON-ifies the error message.

Something along the lines of the following could work (untested):
```rs
#[cfg(feature = "editorconfig")]
let used_config = match is_default_config {
    true => editorconfig::parse(Some(config), &path).context("could not parse editorconfig"),
    false => Ok(Some(config)),
};

tx.send(
    used_config
        .and_then(|used_config| {
            format_file(
                &path,
                used_config.unwrap_or(config),
                range,
                &opt,
                verify_output,
            )
        })
        .map_err(|error| {
            ErrorFileWrapper {
                file: path.display().to_string(),
                error,
            }
            .into()
        }),
)
.unwrap()
```

You may need to convert `parse` to return an `anyhow::Result` instead (like `format_file` does)

Reply:
Done for the second check, unsure about [this line](https://github.com/JohnnyMorganz/StyLua/blob/262dbce6d749fa9e0e2980b4e3a05aef45ce8a3f/src/cli/main.rs#L429) for the first check (stdin), should `map_err` be used twice around here?